pkg/agent/openflow/pipeline.go

```diff
@@ -1417,163 +1755,126 @@ func getIPProtocol(ip net.IP) binding.Protocol {
-			Action().GotoTable(arpResponderTable.GetID()).
-			Cookie(c.cookieAllocator.Request(category).Raw()).
-			Done())
-	}
-	return
-}
-
-// arpSpoofGuardFlow generates the flow to check ARP traffic sent out from local pods interfaces.
-func (c *client) arpSpoofGuardFlow(ifIP net.IP, ifMAC net.HardwareAddr, ifOFPort uint32, category cookie.Category) binding.Flow {
-	return SpoofGuardTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolARP).
+		MatchProtocol(binding.ProtocolARP).
 		MatchInPort(ifOFPort).
 		MatchARPSha(ifMAC).
 		MatchARPSpa(ifIP).
-		Action().GotoTable(arpResponderTable.GetID()).
-		Cookie(c.cookieAllocator.Request(category).Raw()).
+		Action().NextTable().
 		Done()
 }
 
-// sessionAffinityReselectFlow generates the flow which resubmits the service accessing
-// packet back to ServiceLBTable if there is no endpointDNAT flow matched. This
-// case will occur if an Endpoint is removed and is the learned Endpoint
+// Feature: Service
+// Stage: PreRoutingStage
+// Tables: EndpointDNATTable
+// Refactored from:
+//   - func (c *client) sessionAffinityReselectFlow() binding.Flow
+// sessionAffinityReselectFlow generates the flow which resubmits the Service accessing packet back to ServiceLBTable
+// if there is no endpointDNAT flow matched. This case will occur if an Endpoint is removed and is the learned Endpoint
 // selection of the Service.
-func (c *client) sessionAffinityReselectFlow() binding.Flow {
-	return EndpointDNATTable.BuildFlow(priorityLow).
+func (c *featureService) sessionAffinityReselectFlow(category cookie.Category) binding.Flow {
+	return EndpointDNATTable.ofTable.BuildFlow(priorityLow).
+		Cookie(c.cookieAllocator.Request(category).Raw()).
 		MatchRegMark(EpSelectedRegMark).
 		Action().LoadRegMark(EpToSelectRegMark).
-		Action().ResubmitToTable(ServiceLBTable.GetID()).
-		Cookie(c.cookieAllocator.Request(cookie.Service).Raw()).
+		Action().ResubmitToTables(ServiceLBTable.ofTable.GetID()).
 		Done()
 }
 
-// gatewayIPSpoofGuardFlow generates the flow to skip spoof guard checking for traffic sent from gateway interface.
-func (c *client) gatewayIPSpoofGuardFlows(category cookie.Category) []binding.Flow {
+// Feature: PodConnectivity
+// Stage: ValidationStage
+// Tables: SpoofGuardTable
+// Refactored from:
+//   - func (c *client) gatewayIPSpoofGuardFlows(category cookie.Category) []binding.Flow
+// gatewayIPSpoofGuardFlows generates the flow to skip spoof guard checking for packets from the Antrea gateway.
+func (c *featurePodConnectivity) gatewayIPSpoofGuardFlows(category cookie.Category) []binding.Flow {
 	var flows []binding.Flow
-	for _, proto := range c.ipProtocols {
-		nextTable := SpoofGuardTable.GetNext()
-		if proto == binding.ProtocolIPv6 {
-			nextTable = IPv6Table.GetID()
+	for _, ipProtocol := range c.ipProtocols {
+		nextTable := SpoofGuardTable.ofTable.GetNext()
+		if ipProtocol == binding.ProtocolIPv6 {
+			nextTable = IPv6Table.ofTable.GetID()
```

Comment:
Why do you need a hardcode for `nextTable` here, is it possible to generate dyminally by the pipeline?

Reply:
Updated with action `GotoStage`.